Cargo.toml

```diff
@@ -16,6 +16,7 @@ default = ["span"]
 span = []
 
 [dependencies]
+bon = { version = "2.1.0", default-features = false } # 3.2.0 requires rustc 1.59.0
```

Comment:
This pulls in quite a few dependencies for the procedural macro.
Unfortunately, making it optional would turn the code in `fmt` quite messy.

Reply:
oh, if it pulls in a lot, maybe it's better to hand-write the builder. It's really not a lot of trouble, even if it feels like a bit of boilerplate, imo.

Reply:
Will do 👍

Yes, it pulls in `Syn`. (I think it would be helpful to commit `Cargo.lock` to the repository, in part to see those changes in PRs.)

Reply:
> or actually, if the config is defaultable, just don't bother with the builder at all, tbh.

Oh, I missed this comment. Should I remove the builder?

I think it might be convenient to have the defaults `const`-available though.